water/include/water/Dialect/Wave/IR/WaveAttrs.td

```diff
@@ -341,20 +376,29 @@ def WaveIndexMappingAttr : AttrDef<WaveDialect, "WaveIndexMapping"> {
   }];
 
   let parameters = (ins
-    ArrayRefParameter<"WaveSymbolAttr">:$symbol_names,
+    ArrayRefParameter<"::mlir::Attribute">:$symbol_names,
     "::mlir::AffineMap":$start,
     "::mlir::AffineMap":$step,
     "::mlir::AffineMap":$stride
   );
 
   let hasCustomAssemblyFormat = 1;
+  let genVerifyDecl = 0;
 
   let extraClassDeclaration = [{
     /// Get the symbol name at the given index.
     ::llvm::StringRef getSymbolName(unsigned index) {
       auto names = getSymbolNames();
       if (index >= names.size()) return ::llvm::StringRef();
-      return names[index].getName();
+
+      auto name = names[index];
```

Comment:
Nit: expand auto here. Even better if the definition of this entire function is moved to C++